### DISCUSSION TITLE
Timestamp-aware reverse geocoding

### DISCUSSION THREAD

**Discussion by dror3go** [photoprism/photoprism#3437] [Ideas]
When indexing, PhotoPrism makes heavy usage in reverse geocoding: to set time zone, country, title, and keywords (anything else?).

Since this information is time-related, wouldn't it be great if the decision which information to set would use the photo's timestamp?
For example, a specific location might result in business name X for today, but for a picture taken twenty years ago - this business name is probably not relevant since back then there was a different business at that location.

This applies not only for businesses, but also for streets, countries, and maybe other data being returned from the reverse geocoder.

I'm not sure if the geocoder itself provides a way to query the location at a given timestamp, or that it responds with a set of various data, including when each was relevant. Or maybe it's not possible at all? Would love to know.

**Comment by lastzero** [comment]
That is absolutely correct. We have plans to do this. It just doesn't seem to be a very pressing issue, and requires additional resources on our end to implement it (development time in particular, as well as additional servers).

For newly added pictures that have been taken recently, a location data snapshot is automatically created as the location details are permanently cached. This may not be the case for other software, especially if it is cloud-hosted and/or uses commercial APIs like Google Maps, where permanently storing details is not encouraged or explicitly prohibited.

As for the technical implementation, I don't think OSM itself provides an easy way to get historical data - unless you have multiple indexes, e.g. one for each year or every few years. If you find out more, please let us know! :+1:

**Reply by dror3go** [reply]
If it's planned - I'm pleased. Thanks!
